gravityforms/gravityformsakismet
# Add UI refinements to prepare for 1.0 release.

## Description
This PR introduces a number of UI enhancements to the Gravity Forms Akismet add-on to prepare it for the 1.0 release.

Closes gravityforms/backlog#1392.

## Testing Instructions
## UI
Confirm that all of the UI changes request in the aforementioned issue have been addressed and meet the design specifications. This add-on does not have a stylesheet - it inherits all of its settings page styles from Gravity Forms. However, it does seem that some adjustments may be needed.

## QA
1. Install the `master` branch of Gravity Forms, this branch of Gravity Forms Akismet, and the Akismet plugin from the WordPress plugins repository. Enable logging for the add-on on the settings page.
2. Before activating Akismet, activate Gravity Forms and GF Akismet and then visit "Forms". You should see a notice that requirements are missing.
3. Go and active and configure Akismet, then visit Forms -> Settings. If Akismet Integration is enabled, disable it. Go back to "Forms" and confirm that you see a different "missing requirements" notification.
4. Re-enable the Akismet Integration, then create a new form with name, e-mail, and paragraph fields and save your changes.
5. Visit Form Settings -> Akismet. Ensure that you see the Akismet icon in the settings drop-down and the sidebar of the page. Ensure, also, that the "Enabled" field appears as a toggle, and that first name, last name, e-mail, and URL fields appear as drop-downs. Configure the form with the defaults, then save your changes.
6. Preview and submit the form.
7. View the entry details for your submission. Click "Mark as Spam", and confirm that you see a note in the entry indicating the time when it was marked as spam.
8. Review your logs and confirm that you see details about your akismet fields and that they match the values that were submitted.
8. Next, go back to form settings and disable the spam check.
9. Delete your log file.
10. Submit your form again. Review your logs and ensure you do not see log output with the akismet fields.
11. View the form entry. Mark it as spam, and confirm that you _do_ still see the note added to the entry (even though the spam filters are disabled, manually marking something as spam should still show the indication).
12. Check out Gravity Forms 2.4, and review the form settings once more. The "enabled" toggle should be a checkbox. Repeat the form submission steps above with the filters both enabled and disabled.
13. Confirm that there are no errors in the WordPress debug log.

## Screenshots
### Missing requirements notices
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/1407326/123285346-93a1b080-d4d2-11eb-9fb7-bbe9d1b8300c.png">
<img width="1188" alt="image" src="https://user-images.githubusercontent.com/1407326/123285432-a74d1700-d4d2-11eb-8bc0-9338876ec524.png">

### Settings screens
GF 2.4 (checkbox instead of toggle)
<img width="1081" alt="image" src="https://user-images.githubusercontent.com/1407326/123285141-694ff300-d4d2-11eb-879a-7df2e5062dac.png">

GF 2.5 (toggle instead of checkbox)
<img width="936" alt="image" src="https://user-images.githubusercontent.com/1407326/123285050-576e5000-d4d2-11eb-889d-307e513c22ab.png">

### 2.5 Menu icon
<img width="322" alt="image" src="https://user-images.githubusercontent.com/1407326/123285003-4de4e800-d4d2-11eb-88ad-ac79bfd5e5c0.png">

### Entry notes when marked as spam
<img width="824" alt="image" src="https://user-images.githubusercontent.com/1407326/123284932-41608f80-d4d2-11eb-8312-e635ab2d3443.png">
